mods/gen4/formats-data.js

```diff
@@ -1664,7 +1664,7 @@ exports.BattleFormatsData = {
 	},
 	registeel: {
 		inherit: true,
-		randomBattleMoves: ["stealthrock", "ironhead", "curse", "rest", "sleeptalk", "thunderwave", "toxic"],
+		randomBattleMoves: ["stealthrock", "ironhead", "curse", "rest", "sleeptalk", "thunderwave", "toxic", "protect", "seismictoss"],
```

Comment:
Bad sets are already happening, and this should at least make them less frequent. I've repeatedly been given Curse / Iron Head / Toxic / SR. I'm pretty sure what's happening is that exactly one of Rest or Sleep Talk gets selected in the first 4 moves (something like SR / Iron Head / Toxic / Rest), which causes both Rest and Sleep Talk to be rejected, and the movepool is so small that it's unable to reject Curse as a replacement for that last slot even though it wants to. Adding Protect gives you something else to fill the slot with.

Seismic Toss doesn't really help with that problem but it doesn't make it worse either. The worst case that I can think of is that the first 4 moves are Iron Head / Protect / T-Wave / Rest. Protect/Rest/Sleep Talk will get rejected on the first pass and T-Wave will get rejected by Toxic on the second pass, leaving you with Iron Head / SR / Toxic as the first 3 moves and only Curse or S-Toss left to fill the last slot. Both are bad options, but S-Toss is at least no worse than Curse there imo. On the other hand, in the happy case where we aren't forced into a bad set, I think S-Toss is usually gonna be better on any set that doesn't have Curse.

I'm open to suggestions but this is the best solution I could think of that didn't involve adding even worse moves like Earthquake or else completely cutting out either Curse or non-Curse sets (which are both usually really good, so it would be a shame).